core.py

```
@@ -110,146 +124,244 @@ def load_data(data_path: str | Path) -> pd.DataFrame:
+
+    Raises:
+        ValueError: If data validation fails
+    """
+    # Check for NaN in target column
+    if target_column in data.columns:
+        nan_count = data[target_column].isna().sum()
+        if nan_count > 0:
+            raise ValueError(
+                f"Target column '{target_column}' contains {nan_count} NaN values. "
+                f"Please remove or impute these values before training."
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)